src/cascadia/TerminalControl/ControlCore.cpp

```diff
@@ -1601,7 +1601,7 @@ namespace winrt::Microsoft::Terminal::Control::implementation
             _cachedSearchResultRows = winrt::single_threaded_vector<int32_t>(std::move(results));
         }
 
-        return _cachedSearchResultRows;
+        return _cachedSearchResultRows ? _cachedSearchResultRows : winrt::single_threaded_vector<int32_t>();

```

Comment:
i still don't understand. is `null` not an acceptable return value for this function?

is there ever a case where we get here and we didn't (1) have or (2) newly populate the cached rows???

Reply:
I was trying to be extra protective on behalf of the caller here. It's totally possible that this is null - the very first time the find dialog is opened, we immediately ask for the search rows, which there won't be any. `ResetIfStale` will have returned false (there was nothing to do, because there was no query), but we didn't cache some result. 

We could at this point cache this empty vector if you _really_ want.

Reply:
I don't think this can happen, because the initial `_searcher` instance has a 0/empty value for all of its members. `ResetIfStale` should then always return `true` the first time it's called, unless we managed to pass an empty search string, which should not happen either, right?